From 4b337fd7a1c2ca872c6b1596dda1a33ebb7e0332 Mon Sep 17 00:00:00 2001
From: Tony Cheng <tony.cheng@amd.com>
Date: Fri, 16 Sep 2016 23:25:43 -0400
Subject: [PATCH 1098/1722] drm/amd/dal: remove deadcode i2c_aux remove
 keep_power_up_count

- not being called in dal2 either.  keep the counter in object in case we need to revive this feature

Change-Id: I715379af379f037e3810b4e074de80e8287bc50c
Signed-off-by: Tony Cheng <tony.cheng@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 .../drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c    |  1 -
 .../drm/amd/dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c | 17 -----------------
 .../drm/amd/dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c |  1 -
 .../gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c  |  1 -
 .../drm/amd/dal/dc/i2caux/dce80/i2c_hw_engine_dce80.c   | 17 -----------------
 .../drm/amd/dal/dc/i2caux/dce80/i2c_sw_engine_dce80.c   |  1 -
 drivers/gpu/drm/amd/dal/dc/i2caux/engine.h              |  9 ---------
 drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c         | 14 --------------
 drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c       |  1 -
 drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c              | 16 ----------------
 drivers/gpu/drm/amd/dal/include/i2caux_interface.h      |  5 -----
 11 files changed, 83 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
index 8a235289..cb43f83 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
@@ -712,7 +712,6 @@ static const struct aux_engine_funcs aux_engine_funcs = {
 static const struct engine_funcs engine_funcs = {
 	.release_engine = release_engine,
 	.submit_request = dal_aux_engine_submit_request,
-	.keep_power_up_count = dal_i2caux_keep_power_up_count,
 	.get_engine_type = dal_aux_engine_get_engine_type,
 	.acquire = dal_aux_engine_acquire,
 };
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c
index 7f0aa8d..dbffe9c 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c
@@ -177,22 +177,6 @@ static void release_engine(
 		disable_i2c_hw_engine(hw_engine);
 }
 
-static void keep_power_up_count(
-	struct engine *engine,
-	bool keep_power_up)
-{
-	struct i2c_hw_engine_dce110 *hw_engine = FROM_ENGINE(engine);
-
-	if (keep_power_up)
-		++hw_engine->engine_keep_power_up_count;
-	else {
-		--hw_engine->engine_keep_power_up_count;
-
-		if (!hw_engine->engine_keep_power_up_count)
-			disable_i2c_hw_engine(hw_engine);
-	}
-}
-
 static bool setup_engine(
 	struct i2c_engine *i2c_engine)
 {
@@ -844,7 +828,6 @@ static const struct i2c_engine_funcs i2c_engine_funcs = {
 
 static const struct engine_funcs engine_funcs = {
 	.release_engine = release_engine,
-	.keep_power_up_count = keep_power_up_count,
 	.get_engine_type = dal_i2c_hw_engine_get_engine_type,
 	.acquire = dal_i2c_engine_acquire,
 	.submit_request = dal_i2c_hw_engine_submit_request,
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c
index 0a339b8..996813d 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c
@@ -116,7 +116,6 @@ static const struct engine_funcs engine_funcs = {
 	.get_engine_type = dal_i2c_sw_engine_get_engine_type,
 	.acquire = dal_i2c_engine_acquire,
 	.submit_request = dal_i2c_sw_engine_submit_request,
-	.keep_power_up_count = dal_i2caux_keep_power_up_count,
 };
 
 static bool construct(
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c
index 4b81f67..34a37e1 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c
@@ -665,7 +665,6 @@ static const struct aux_engine_funcs aux_engine_funcs = {
 static const struct engine_funcs engine_funcs = {
 	.release_engine = release_engine,
 	.submit_request = dal_aux_engine_submit_request,
-	.keep_power_up_count = dal_i2caux_keep_power_up_count,
 	.get_engine_type = dal_aux_engine_get_engine_type,
 	.acquire = dal_aux_engine_acquire,
 };
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/i2c_hw_engine_dce80.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/i2c_hw_engine_dce80.c
index 07f6899..423c38a 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/i2c_hw_engine_dce80.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/i2c_hw_engine_dce80.c
@@ -177,22 +177,6 @@ static void release_engine(
 		disable_i2c_hw_engine(hw_engine);
 }
 
-static void keep_power_up_count(
-	struct engine *engine,
-	bool keep_power_up)
-{
-	struct i2c_hw_engine_dce80 *hw_engine = FROM_ENGINE(engine);
-
-	if (keep_power_up)
-		++hw_engine->engine_keep_power_up_count;
-	else {
-		--hw_engine->engine_keep_power_up_count;
-
-		if (!hw_engine->engine_keep_power_up_count)
-			disable_i2c_hw_engine(hw_engine);
-	}
-}
-
 static void destruct(
 	struct i2c_hw_engine_dce80 *engine)
 {
@@ -826,7 +810,6 @@ static const struct i2c_engine_funcs i2c_engine_funcs = {
 
 static const struct engine_funcs engine_funcs = {
 	.release_engine = release_engine,
-	.keep_power_up_count = keep_power_up_count,
 	.get_engine_type = dal_i2c_hw_engine_get_engine_type,
 	.acquire = dal_i2c_engine_acquire,
 	.submit_request = dal_i2c_hw_engine_submit_request,
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/i2c_sw_engine_dce80.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/i2c_sw_engine_dce80.c
index 5f2f298..804a326 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/i2c_sw_engine_dce80.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/i2c_sw_engine_dce80.c
@@ -131,7 +131,6 @@ static const struct engine_funcs engine_funcs = {
 	.get_engine_type = dal_i2c_sw_engine_get_engine_type,
 	.acquire = dal_i2c_engine_acquire,
 	.submit_request = dal_i2c_sw_engine_submit_request,
-	.keep_power_up_count = dal_i2caux_keep_power_up_count,
 };
 
 static bool construct(
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/engine.h b/drivers/gpu/drm/amd/dal/dc/i2caux/engine.h
index dec35f0..76fe2df 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/engine.h
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/engine.h
@@ -100,11 +100,6 @@ struct engine_funcs {
 		struct engine *engine,
 		struct i2caux_transaction_request *request,
 		bool middle_of_transaction);
-	/* [anaumov] Actually, following method is meaningful
-	 * only in I2C HW engines */
-	void (*keep_power_up_count)(
-		struct engine *engine,
-		bool keep_power_up);
 	void (*release_engine)(
 		struct engine *engine);
 };
@@ -122,8 +117,4 @@ bool dal_i2caux_construct_engine(
 void dal_i2caux_destruct_engine(
 	struct engine *engine);
 
-void dal_i2caux_keep_power_up_count(
-	struct engine *engine,
-	bool keep_power_up);
-
 #endif
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c b/drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c
index 9c472b9..09da813 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c
@@ -36,20 +36,6 @@
 
 #include "engine.h"
 
-/*
- * Post-requisites: headers required by this unit
- */
-
-/*
- * This unit
- */
-void dal_i2caux_keep_power_up_count(
-	struct engine *engine,
-	bool keep_power_up)
-{
-
-}
-
 bool dal_i2caux_construct_engine(
 	struct engine *engine,
 	struct dc_context *ctx)
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c
index a7d5a2a..5a9455c 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c
@@ -568,7 +568,6 @@ static const struct engine_funcs engine_funcs = {
 	.get_engine_type = dal_i2c_sw_engine_get_engine_type,
 	.acquire = dal_i2c_engine_acquire,
 	.submit_request = dal_i2c_sw_engine_submit_request,
-	.keep_power_up_count = dal_i2caux_keep_power_up_count,
 };
 
 bool dal_i2c_sw_engine_construct(
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c b/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c
index 6d1d36b..73db3d4 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c
@@ -289,22 +289,6 @@ static bool get_hw_supported_ddc_line(
 	return true;
 }
 
-void dal_i2caux_keep_engine_power_up(
-	struct i2caux *i2caux,
-	struct ddc *ddc,
-	bool keep_power_up)
-{
-	enum gpio_ddc_line line;
-	struct i2c_engine *engine;
-
-	if (!get_hw_supported_ddc_line(ddc, &line))
-		return;
-
-	engine = i2caux->i2c_hw_engines[line];
-
-	engine->base.funcs->keep_power_up_count(&engine->base, keep_power_up);
-}
-
 bool dal_i2caux_start_gtc_sync(
 	struct i2caux *i2caux,
 	struct ddc *ddc)
diff --git a/drivers/gpu/drm/amd/dal/include/i2caux_interface.h b/drivers/gpu/drm/amd/dal/include/i2caux_interface.h
index 17c7768..6e44f19 100644
--- a/drivers/gpu/drm/amd/dal/include/i2caux_interface.h
+++ b/drivers/gpu/drm/amd/dal/include/i2caux_interface.h
@@ -79,11 +79,6 @@ bool dal_i2caux_submit_aux_command(
 	struct ddc *ddc,
 	struct aux_command *cmd);
 
-void dal_i2caux_keep_engine_power_up(
-	struct i2caux *i2caux,
-	struct ddc *ddc,
-	bool keep_power_up);
-
 bool dal_i2caux_start_gtc_sync(
 	struct i2caux *i2caux,
 	struct ddc *ddc);
-- 
2.7.4

